service/controller/resource/nodepool/create_deployment_uninitialized.go

```diff
@@ -90,11 +91,18 @@ func (r *Resource) deploymentUninitializedTransition(ctx context.Context, obj in
 		return currentState, microerror.Mask(err)
 	}
 
-	deploymentIsOutOfDate, nodesNeedToBeRolled, err := r.deploymentIsOutOfDate(ctx, currentDeployment, desiredDeployment)
+	deploymentIsOutOfDate, err := template.IsOutOfDate(currentDeployment, desiredDeployment)
 	if err != nil {
 		return currentState, microerror.Mask(err)
 	}
 
+	nodesNeedToBeRolled, err := template.NeedToRolloutNodes(currentDeployment, desiredDeployment)
```

Comment:
I'm fine with this now, but would it make sense to have some kind of subtype or something so that nodes are not mixed with template? I feel they are conceptually two different things. I can't think of anything better right now, but maybe just as a food for a thought?

Reply:
Yeah, I agree but I don't know how to solve it either. Thinking out loud here, but we could make `IsOutOfDate` return an slice of strings with the name of the parameters that have changed. So you know is out of date if the length of the slice is greater than 0. Then, clients of this function like the `nodepool` handler can make decision based on the slice. Does it only contain the `scaling` field? Then I need to roll out nodes. But this way the node concept is removed from `template`.

Reply:
I pushed changes following that idea. Let me know what you think.